tests/unit/iotics/lib/identity/api/test_advanced_api.py

```diff
@@ -303,3 +314,81 @@ def test_can_revoke_delegation_proof(base_doc_issuer, valid_key_pair, get_api_ca
     get_api_call(api)(deleg_name, revoked=True, doc_owner_issuer=base_doc_issuer, doc_owner_key_pair=valid_key_pair)
     updated_doc = resolver_client.docs[base_doc_issuer.did]
     assert doc_deleg_set(updated_doc)[deleg_name].revoked
+
+
+@pytest.mark.parametrize('get_delegation_method', (lambda api: api.add_control_delegation_proof_to_document,
+                                                   lambda api: api.add_authentication_delegation_proof_to_document))
+def test_can_not_reuse_not_reusable_delegation_proof(get_delegation_method):
+    twin1_secrets, twin1_issuer, twin1_doc = get_new_document('#Twin1')
+    twin2_secrets, twin2_issuer, twin2_doc = get_new_document('#Twin2')
+    agent_secrets, agent1_issuer, agent_doc = get_new_document('#Agent1')
+    resolver_client = ResolverClientTest(docs={
+        twin1_doc.did: twin1_doc,
+        twin2_doc.did: twin2_doc,
+        agent_doc.did: agent_doc,
+    })
+    api = AdvancedIdentityRegisterApi(resolver_client)
+
+    # Create an agent not reusable delegation proof (targeting a single delegating document: twin1)
+    _, proof = AdvancedIdentityLocalApi.create_delegation_proof(delegating_issuer=twin1_issuer,
+                                                                subject_doc=agent_doc,
+                                                                subject_secrets=agent_secrets)
+
+    # Add proof to twin1 document as a delegation proof - the document is valid
+    get_delegation_method(api)(
+        proof=proof,
+        subject_issuer=agent1_issuer,
+        delegation_name='#Deleg1',
+        doc_owner_issuer=twin1_issuer,
+        doc_owner_key_pair=KeyPairSecretsHelper.get_key_pair(twin1_secrets),
+    )
+    assert is_validator_run_success(api.validate_register_document, resolver_client.docs[twin1_doc.did])
+
+    # Add proof to twin1 document as a delegation proof - the document is invalid
+    get_delegation_method(api)(
+        proof=proof,
+        subject_issuer=agent1_issuer,
+        delegation_name='#Deleg1',
```

Comment:
update name just to be more clear its a second one

Reply:
https://github.com/Iotic-Labs/iotics-identity-py/pull/17/commits/828cce4f4ea4c9f180cf8b470c20df6b2cb81349